src/Simulation/Common/Utils.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Quantum.Simulation.Core;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.Quantum.Simulation.Common
+{
+    public class CommonUtils
+    {
+        /// <summary>
+        /// Removes PauliI terms from observable and corresponding qubits from qubits. 
+        /// Returns the observable description that is equivalent to the original one, but has no PauliI terms
+        /// </summary>
+        public static void PruneObservable(IQArray<Pauli> observable, IQArray<Qubit> qubits, out QArray<Pauli> prunnedObservable, out QArray<Qubit> prunnedQubits)
```

Comment:
No, I don't think would be good. This function removes extras from both observable and qubits. Having it on one of them (observable) will be asymmetrical and misleading. Otherwise I probably would not have strong feelings either way. Both work and do not affect performance.

---
In reply to: [348340204](https://github.com/microsoft/qsharp-runtime/pull/82#discussion_r348340204) [](ancestors = 348340204)

Reply:
It does look like "pruned" has been misspelled here, though?